### PR TITLE
[jaeger] Adds ability to set enableServiceLinks to false

### PR DESCRIPTION
**Pull request by koalalorenzo**
#### What this PR does
Attempt to fix: https://github.com/jaegertracing/jaeger/issues/1986 and https://github.com/prometheus/prometheus/issues/7286

#### Checklist

- [x] [DCO](https://github.com/jaegertracing/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [X] Commits are [GPG signed](https://docs.github.com/en/github/authenticating-to-github/about-commit-signature-verification)
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (`[jaeger]` or `[jaeger-operator]`)

### CODE REVIEW

File: charts/jaeger/values.yaml [hunk 1]
```diff
@@ -200,6 +200,9 @@ agent:
       # type: RollingUpdate
       # rollingUpdate:
       #   maxUnavailable: 1
+  # Set to false if your release is called `jaeger`.
+  # Read More: https://github.com/jaegertracing/jaeger/issues/1986
```
Comment (naseemkullah) [inline charts/jaeger/values.yaml:204]: https://github.com/jaegertracing/jaeger/issues/1986  concerns the query component was worked around via https://github.com/jaegertracing/helm-charts/pull/12, can you plz elaborate on what issue this solves or how to reproduce? Thanks!

**Review by naseemkullah — CHANGES_REQUESTED** [review]

**Comment by naseemkullah** [comment]
So I just started seeing this in my setup as well, i recently moved jaeger into the same namespace as prometheus, and prometheus emits:

```
{"caller":"main.go:560","err":"unable to get tracing config from environment: cannot obtain reporter config from env: cannot parse env var JAEGER_AGENT_PORT=udp://11.2.3.4:5775: strconv.ParseInt: parsing \"udp://1.2.3.4:5775\": invalid syntax","level":"error","msg":"Unable to init tracing","ts":"2021-02-09T16:35:46.791Z"}
````

**Comment by pavelnikolov** [comment]
I've never added the protocol prefix before. Usually use `<ip>:<port>`

**Comment by naseemkullah** [comment]
I think the prometheus server  would have to have `JAEGER_AGENT_PORT` in its env (as done for jaeger query) or `enableServiceLinks` disabled, WDYT @koalalorenzo ?

**Comment by naseemkullah** [comment]
(I've since removed jaeger-agent from our infra in favour of otel-collector agent)

File: charts/jaeger/values.yaml (hunk 1 above)
Comment (koalalorenzo) [reply charts/jaeger/values.yaml:204]: This is supposed to resolve a problem related to exposing jaeger service port via an env variable, causing major issues if the chart name is exactly "jaeger".

**Comment by koalalorenzo** [comment]
@naseemkullah I am not having a good experience with jaeger.

Our **fix** consists into temporary change the `jaeger` name and explore other solutions.
To "fix" it we needed to change `enableServiceLinks` in the manifests of jaeger, as described in [this issue](https://github.com/jaegertracing/jaeger/issues/1986#issuecomment-775183288), so this PR is about that anyway.

**Comment by naseemkullah** [comment]
> @naseemkullah I am not having a good experience with jaeger.
> 
> 
> 
> Our **fix** consists into temporary change the `jaeger` name and explore other solutions to replace entirely jaeger
> 
> 
> 
> To "fix" it we needed to change `enableServiceLinks` in the manifests of jaeger, as described in [this issue](https://github.com/jaegertracing/jaeger/issues/1986#issuecomment-775183288), so this PR is about that anyway.

That issue is about the jaeger query, (I would know I created the issue), I also have a release called jaeger. Please see my comment about Prometheus emitting a log. What issue are you facing? Which component is producing an error?

File: charts/jaeger/values.yaml (hunk 1 above)
Comment (naseemkullah) [inline charts/jaeger/values.yaml:204]:
```suggestion
```

The issue link has been worked around, and for prometheus there are other workarounds such as adding `JAEGER_AGENT_PORT` in prometheus server container's env.

That aside adding the enableServiceLinks bool to the agent and any other components is a welcome addition!

**Review by naseemkullah — CHANGES_REQUESTED** [review]

**Comment by koalalorenzo** [comment]
> It sound like you are facing issues with prometheus erroring out about JAEGER_AGENT_PORT but you would have to confirm.

Yes correct.

**Review by naseemkullah — APPROVED** [review]

**Comment by naseemkullah** [comment]
Thanks @koalalorenzo! lgtm, but all checks will need to pass before merging though.

**Comment by koalalorenzo** [comment]
I think that the commit from GitHub adds the issue as it is not signed nor signed with a GPG :(

**Comment by naseemkullah** [comment]
Hi @koalalorenzo GPG seems ok, its the DCO that needs to be signed in this case and conflicts need to be resolved to get this merged.

**Comment by jonbcampos-alto** [comment]
+1

**Comment by mehta-ankit** [comment]
> Hi @koalalorenzo GPG seems ok, its the DCO that needs to be signed in this case and conflicts need to be resolved to get this merged.

@koalalorenzo If you are still looking to push this through. This seems to be the issue:-
image

**Pull request closed without merging**